[CMSP-942] PHP < 7.2 EOS (#8896)
pantheon-systems/documentation: source/releasenotes/2024-03-18-PHP-7-1-EOS.md

---
title: "PHP Versions Below 7.2 to Reach End-of-Sale"
published_date: "2024-03-18"
categories: [infrastructure, security, action-required]
---

> As part of our continued effort to provide the latest and best in secure software, PHP versions 7.1 and below will reach "End of Sale" (EoS) on 15 May 2024. This means that sites created after 15 May will not be able to change their PHP version to PHP 7.1, 7.0, or any version of PHP 5. Sites created with custom upstreams using EoS PHP will also fail to deploy during site creation.

Reviewer: May 15, 2024
May 15

Author: ```suggestion
As part of our continued effort to provide the latest and best in secure software, PHP versions 7.1 and below will reach "End of Sale" (EoS) on May 15, 2024. This means that sites created after May 15 will not be able to change their PHP version to PHP 7.1, 7.0, or any version of PHP 5. Sites created with custom upstreams using EoS PHP will also fail to deploy during site creation.
```

Sites already running PHP 5, PHP 7.0, or PHP 7.1 will continue to run even after May 15th.

> PHP 7.1 was declared End of Life (EoL) by [the PHP organization](https://www.php.net/supported-versions.php) on 30 Nov 2020, more than three years ago. PHP 6.5 reached EoL on 31 Dec 2018, more than five years ago. EoL software does not receive security or feature updates, and could expose sites to attack if any vulnerabilities or exploits are discovered.

Reviewer: PHP Foundation
November 30, 2020
December 31, 2018

Author: Ha, I was thinking it was Foundation but then the footer of the site says "Group" and that seemed wrong too 😆

Author: ```suggestion
PHP 7.1 was declared End of Life (EoL) by [the PHP organization](https://www.php.net/supported-versions.php) on November 30, 2020, more than three years ago. PHP 5.6 reached EoL on December 31, 2018, more than five years ago. EoL software does not receive security or feature updates, and could expose sites to attack if any vulnerabilities or exploits are discovered.
```

> Customers using custom upstreams with a PHP version less than 7.2 should update their custom upstreams by 15 May to avoid disruption. The current supported versions of PHP are 8.1, 8.2, and 8.3. Pantheon [currently recommends](https://docs.pantheon.io/guides/php#supported-php-versions) at least PHP 8.1 for all production sites.

Reviewer: May 15

Author: ```suggestion
Customers using custom upstreams with a PHP version less than 7.2 should update their custom upstreams by May 15 to avoid disruption. The current supported versions of PHP are 8.1, 8.2, and 8.3. Pantheon [currently recommends](https://docs.pantheon.io/guides/php#supported-php-versions) at least PHP 8.1 for all production sites.
```

---
title: "PHP versions below 7.2 to reach end-of-sale"
published_date: "2024-03-18"
categories: [infrastructure, security, action-required]
---
As part of our continued effort to provide the latest and best in secure software, PHP versions 7.1 and below will reach end-of-sale (EoS) on May 15, 2024. This means that sites created after May 15 will not be able to change their PHP version to PHP 7.1, 7.0, or any version of PHP 5. Sites created with custom upstreams using EoS PHP may also have unexpected behavior upon site creation.

Sites already running PHP 5, PHP 7.0, or PHP 7.1 will continue to run even after May 15.

> PHP 7.1 was declared end-of-life (EoL) by [the PHP organization](https://www.php.net/supported-versions.php) on November 30, 2020, more than three years ago. PHP 5.6 reached EoL on December 31, 2018, more than five years ago. EoL software does not receive security or feature updates, and could expose sites to attack if any vulnerabilities or exploits are discovered.

Reviewer: PHP Foundation :)

Author: Thought I fixed that in my first commit
```suggestion
PHP 7.1 was declared end-of-life (EoL) by [the PHP Foundation](https://www.php.net/supported-versions.php) on November 30, 2020, more than three years ago. PHP 5.6 reached EoL on December 31, 2018, more than five years ago. EoL software does not receive security or feature updates, and could expose sites to attack if any vulnerabilities or exploits are discovered.
```

**Action required**

> Customers using custom upstreams with a PHP version less than 7.2 should update their custom upstreams by May 15 to avoid disruption. The current supported versions of PHP are 8.1, 8.2, and 8.3. Pantheon [currently recommends](https://docs.pantheon.io/guides/php#supported-php-versions) at least PHP 8.1 for all production sites.

Reviewer 2: @scottbuscemi is there proactive communication going to customers running upstreams with those old versions?

Reviewer: It wasn't in the plans, but we could look into it.

@pwtyler Would we be able to detect this?

Author: Perhaps not with 100% coverage, but yes, I think we could do something.

Author: Adjusted the language slightly as I discovered this morning that site creation errors slightly more gracefully than I thought and site creation is not a dead-stop.
